src/EditorFeatures/CSharp/StringCopyPaste/StringCopyPasteHelpers.cs

```diff
@@ -0,0 +1,622 @@
+        {
+            if (end > start && text[end - 1] == '8')
+                end--;
+            if (end > start && text[end - 1] is 'u' or 'U')
+                end--;
+            return end;
+        }
+
+        private static ImmutableArray<TextSpan> GetInterpolatedStringTextContentSpans(
+            SourceText text, InterpolatedStringExpressionSyntax interpolatedString,
+            out int delimiterQuoteCount, out int delimiterDollarCount)
+        {
+            // Interpolated string.  Normal, verbatim, or raw.
+            //
+            // Skip past the leading and trailing delimiters.
+            var start = interpolatedString.SpanStart;
+            while (start < text.Length && text[start] is '@' or '$')
+                start++;
+            delimiterDollarCount = start - interpolatedString.SpanStart;
+
+            var position = start;
+            while (start < interpolatedString.StringStartToken.Span.End && text[start] == '"')
+                start++;
+            delimiterQuoteCount = start - position;
+
+            var end = interpolatedString.Span.End;
+
+            end = SkipU8Suffix(text, start, end);
+            while (end > interpolatedString.StringEndToken.Span.Start && text[end - 1] == '"')
+                end--;
+
+            using var result = TemporaryArray<TextSpan>.Empty;
+            var currentPosition = start;
+            for (var i = 0; i < interpolatedString.Contents.Count; i++)
+            {
+                var content = interpolatedString.Contents[i];
+                if (content is InterpolationSyntax)
+                {
+                    result.Add(TextSpan.FromBounds(currentPosition, content.SpanStart));
+                    currentPosition = content.Span.End;
+                }
+            }
+
+            // Then, once through the body, add a final span from the end of the last interpolation to the end delimiter.
+            result.Add(TextSpan.FromBounds(currentPosition, end));
+            return result.ToImmutableAndClear();
+        }
+
+        private static TextSpan GetStringLiteralTextContentSpan(SourceText text, LiteralExpressionSyntax literal, out int delimiterQuoteCount)
+        {
+            // simple string literal (normal, verbatim or raw).
+            //
+            // Skip past the leading and trailing delimiters and add the span in between.
+            if (IsRawStringLiteral(literal))
+            {
+                var start = literal.SpanStart;
+                while (start < text.Length && text[start] == '"')
+                    start++;
+                delimiterQuoteCount = start - literal.SpanStart;
+
+                var end = literal.Span.End;
+
+                end = SkipU8Suffix(text, start, end);
+                while (end > start && text[end - 1] == '"')
+                    end--;
+
+                return TextSpan.FromBounds(start, end);
+            }
+            else
+            {
+                var start = literal.SpanStart;
```

Comment:
These two blocks do the same thing.. either this one can be simplified to not loop, or they could just be combined, or at worst, skip the `@` check for raw strings (though not sure that is worth bothering)

Reply:
the two blocks are subtly different (ignoring even the '@' difference). With normal strings, we just peel off the *first* quote at the start/end.  With raw-strings we peel off *all* the quotes at the start/end.  We don't want the latter behavior with normal strings because a normal string can be something like `@"""""foo"""""`.  In this case, the content span is ``@"[|""""foo""""|]"` whereas with raw strings the content span of the nearly equivalent text would be `"""""[|foo|]"""""` :)

Reply:
Yep, you're right.. this block says `if` not `while`.. my bad.

Reply:
i added a commentin the code about this.